quinn-proto/src/endpoint.rs

```diff
@@ -84,10 +84,34 @@ impl Endpoint {
 
     /// Get the next packet to transmit
     #[must_use]
-    pub fn poll_transmit(&mut self) -> Option<Transmit> {
+    pub fn deque_transmit(&mut self) -> Option<Transmit> {
         self.transmits.pop_front()
     }
 
+    /// Queues a packet for transmission
+    pub fn queue_transmit(&mut self, transmit: Transmit) {
+        self.transmits.push_back(transmit)
+    }
+
+    /// Drains a range of transmittable packets
+    pub fn drain_transmits(&mut self, range: RangeTo<usize>) {
+        self.transmits.drain(range);
+    }
+
+    /// Returns a batch of transmittable packets
+    pub fn transmit_batch(&self, batch: usize) -> &[Transmit] {
+        if self.transmits.len() >= batch {
+            &self.transmits.as_slices().0[..batch]
```

Comment:
This may panic, as there's no guarantee that the first slice of a `VecDeque` contains its full length.